ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixClusterManager.java

```diff
@@ -702,6 +702,26 @@ public String getResourceNameForPartition(AmbryPartition partition) {
       return replicas;
     }
 
+    @Override
+    public Map<ReplicaState, List<AmbryReplica>> getReplicaIdsByStates(AmbryPartition partition,
+        Set<ReplicaState> states, String dcName) {
+      Map<ReplicaState, List<AmbryReplica>> replicasByState = new HashMap<>();
+      for (DcInfo dcInfo : dcToDcInfo.values()) {
+        String dc = dcInfo.dcName;
+        if (dcName == null || dcName.equals(dc)) {
+          Map<ReplicaState, List<AmbryReplica>> replicaStateMap =
+              dcInfo.clusterChangeHandler.getSnapshotOfReplicaStates(partition);
```

Comment:
Could potentially have this method take in a map and allowed states as an argument and update the map directly to avoid having to iterate/copy again.

Reply:
yeah, fair point. I'll make a change